src/validation.h

```diff
@@ -1142,6 +1143,7 @@ class ChainstateManager
 
     /** Check whether we are doing an initial block download (synchronizing from disk or network) */
     bool IsInitialBlockDownload() const;
+    void UpdateCachedChaintipRecent() EXCLUSIVE_LOCKS_REQUIRED(cs_main);
```

Comment:
should be const and needs some comment (and I'd specialize it to just return the value instead of mutating the state, we can do that in the `SetTip` method instead)
```suggestion
    /** Check whether the active chain tip exists, has enough work, and is recent. */
    bool IsTipRecent() const EXCLUSIVE_LOCKS_REQUIRED(cs_main);
```